### PR TITLE
[flycast][game controller config] Sets Flycast joystick configuration automatically.

### PR DESCRIPTION
**Pull request by Langerz82**
SET in "/storage/.config/emuelec/emuelec.conf":
```
flycast_auto_gamepad=1
```
Works much like the dolphin auto joy config with a little difference. The hat values are manually obtained through an array. For different controllers they may be different. If so the user should press select for the menu and manually map the dpad values for there controller.
The open "/storage/.config/flycast/mappings/SDL_(controller name).cfg" and get the dpad records and paste them here so I can update the script.

Thanks.

Old text info:

> Works very similarly to set_dolphin_joy.sh with a bit of a difference.
The hat values for my PS3 Controller had to be mapped with an associative array and I'm not sure the values would be the same for different controllers. If so the player can map them by pressing select and going through the settings menu and manually setting the dpad values. Then the user can open the file "/storage/.config/flycast/mappings/SDL_[controller name].cfg" and post the dpad values here I will be able to update the script.
I'm hoping that my hat values will be the same for others, but I am not sure.

### CODE REVIEW

**Comment by Langerz82** [comment]
How do I change this into a draft? I made a mistake.

**Comment by shantigilbert** [comment]
> How do I change this into a draft? I made a mistake.

Don't worry about, I have to merge it for it to be live

**Review by shantigilbert — CHANGES_REQUESTED** [review]
What settings from emu.cfg are needed for this? please remove all the others.

File: packages/sx05re/emulators/flycastsa/package.mk [hunk 1]
```diff
@@ -20,7 +20,14 @@ PKG_CMAKE_OPTS_TARGET+="-DUSE_GLES=ON -DUSE_VULKAN=OFF"
 }
 
 makeinstall_target() {
+	mkdir -p $INSTALL/usr/config/flycast
+	mkdir -p $INSTALL/usr/config/flycast/mappings
```
Comment (shantigilbert) [inline packages/sx05re/emulators/flycastsa/package.mk:24]: Is this needed?

**Comment by Langerz82** [comment]
They are need if the user uses the flycast auto joystick config, but I can move the commands so it checks first and creates them if needed in set_flycast_joy.sh. Regarding flycast config: I will remove them and leave only ones that are needed if any at all.
Give me a few hours to update code.

File: packages/sx05re/emulators/flycastsa/package.mk (hunk 1 above)
Comment (Langerz82) [reply packages/sx05re/emulators/flycastsa/package.mk]: I made the mkdirs in dolphin.sh, yes they are needed if you use gamepad auto detection.

**Comment by Langerz82** [comment]
The detection code in devices needs some more work done, it has bugs!

**Review by shantigilbert — APPROVED** [review]

**Comment by Langerz82** [comment]
I tested it with two game controllers, I think it needs more testing from others.

**Pull request closed without merging**